src/components/tree/phyloTree/grid.js

```diff
@@ -177,10 +102,112 @@ const computeXGridPoints = (xmin, xmax, layout, distanceMeasure, minorTicks, pxA
   return {majorGridPoints, minorGridPoints};
 };
 
+/**
+ * Calculate the spacing between Major and Minor grid points. This is computed via a
+ * heuristic which takes into account (a) the available space (pixels) and (b) the
+ * time range to display.
+ * As major grid lines are (usually) labelled, we wish these to represent a consistent
+ * spacing of time, e.g. "3 months" or "7 years". Note that this means the actual time between
+ * grids may be very slightly different, as months, years etc can have different numbers of days.
+ * @param {numeric} timeRange numeric date range in current view (between right-most tip & left-most node)
+ * @param {numeric} pxAvailable number of pixels available
+ * @returns {object}
+ */
+const calculateTemporalGridSeperation = (timeRange, pxAvailable) => {
+  const [majorStep, minorStep] = [{unit: "DAY", n: 1}, {unit: "DAY", n: 0}];
+
+  /* in general, we find that 1 major point for every ~100px works well
+  for wider displays we shift up to 150px then 200px */
+  const minPxBetweenMajorGrid = (pxAvailable < 1200 ? 120 : 150);
+  const timeBetweenMajorGrids = timeRange/(Math.floor(pxAvailable / minPxBetweenMajorGrid));
+  const levels = {CENTURY: 100, DECADE: 10, YEAR: 1, MONTH: 1/12, WEEK: 1/52, DAY: 1/365};
+  const levelsKeys = Object.keys(levels);
+
+  /* calculate the best unit of time to fit into the allowed range */
+  majorStep.unit = "DAY"; // fallback value
+  for (let i=0; i<levelsKeys.length-1; i++) {
+    if (timeBetweenMajorGrids > levels[levelsKeys[i]]) {
+      majorStep.unit = levelsKeys[i];
+      break;
+    }
+  }
+  /* how many of those "units" should ideally fit into each major grid separation? */
+  majorStep.n = Math.floor(timeBetweenMajorGrids/levels[majorStep.unit]) || 1;
+
+  /* Calculate best unit of time for the minor grid spacing */
+  if (majorStep.n > 1 || majorStep.unit === "DAY") {
+    minorStep.unit = majorStep.unit;
+  } else {
+    minorStep.unit = levelsKeys[levelsKeys.indexOf(majorStep.unit)+1];
+  }
+  /* how many of those "units" should form the separation of the minor grids? */
+  const majorSpacing = majorStep.n * levels[majorStep.unit];
+  minorStep.n = Math.ceil(levels[minorStep.unit]/majorSpacing);
+
+  // XXX TODO remove following lines before merge of PR
```

Comment:
TODO - these console logs left in as they may be useful for review. Should be taken out before merge.